src/engine/N3Base/N3FXPartBillBoard.cpp

```diff
@@ -200,6 +200,14 @@ bool CN3FXPartBillBoard::Load(HANDLE hFile) {
         ReadFile(hFile, &m_mtxRot, sizeof(m_mtxRot), &dwRWC, NULL);
     }
 
+    if (m_iVersion >= 6) {
+        ReadFile(hFile, &m_bRotationRate, sizeof(bool), &dwRWC, NULL);
+    }
+
+    if (m_iVersion >= 7) {
+        ReadFile(hFile, &m_bOnScreen, sizeof(bool), &dwRWC, NULL); // is this on screen bool?
```

Comment:
Unless I'm crazy, I think these 2 are mixed up?

Reply:
Could be like I said I don't have pdbs or asm/reversing experience. This is half of the pr the other was somewhere else to correct few issues. If you want I Can try to search for the src and go over it together?

Reply:
It's a little hard to tell without any implementation behaviour, but based on naming alone (specifically the rotation one), the way I've named them both personally fits with them the other way around (which tracks with their actual implementation behaviour), which makes me think you possibly accidentally mixed up the order here. It's just a little hard to tell from this alone though if you accidentally mixed up the order or the names perhaps aren't the clearest.

Reply:
Probably could have mixed them the other way around very highly you're right. For the naming they are the original naming convention except for versioning which I named myself OFC. I'll try to get them today so you can take a look at it.

Reply:
Bool on screen came later so you must be right...

Reply:
I now understand the order you're pulling them from is the order in the leaked project files. This doesn't mean that they're handled in that order; these project files don't care about the versioning whatsoever, so they're not even currently output in a strict version order, meaning they don't really have any reason to stick to that and throw them in at the bottom.

Now that I understand the reasoning behind the names (leaked effect project files) and why you put them in that order (appearance in the saved project file, which is no guarantee of order in the binary), I can definitely say that they're backwards without question here.